docs/DeveloperGuide.adoc

```diff
@@ -1142,6 +1142,105 @@ These instructions only provide a starting point for testers to work on; testers
+`<xpValueByDay>200</xpValueByDay>` +
+`<numTaskCompletedByDay>5</numTaskCompletedByDay>` +
+`<nextWeekBreakPoint>dd-mm-yy 0000</nextWeekBreakPoint>`(where dd-mm-yy is today's date + 7 days) +
+`<xpValueByWeek>300</xpValueByWeek>` +
+`<numTaskCompletedByWeek>12</numTaskCompletedByWeek>` +
+.. Test case:
+... `achievements today`
+... set the system time of the testing computer to be tomorrow(one day after today)
+... `achievements today` +
+Expected: Before system time change, `achievements today` shows "Daily achievements from [today's date]:", today's
+xp to be 200, tasks completed to be 5. After system time change, `achievements today` shows "Daily achievements from
+[tomorrow's date]:", this day's xp to be 0, tasks completed to be 0. All-time xp value remains 400, tasks completed
+remains 14. This week's xp value remains 300, tasks completed remains 12.
+
+.. Test case:
+... `achievements this week`
+... set the system time of the testing computer to be next week(7 days after today)
+... `achievements this week`
+... `achievements today` +
+Expected: Before system time change, `achievements this week` shows "Weekly achievements from [today's date]:",
+this week's xp to be 300, tasks completed to be 12. After system time change, `achievements this week` shows "Weekly
+achievements from [next week's date]:", this week's xp to be 0, tasks completed to be 0. Now, `achievements today`
+shows "Daily achievements from [next week's date]:", this day's xp to be 0, tasks completed to be 0.
+
+. Detection of invalid/corrupted achievement information from storage +
+ +
+change the specified fields in data/taskmanager.xml(or other storage path) to:
+.. Test case: `<displayOption>5</displayOption>` (invalid display option)
+.. Test case: `<xp>59</xp>` `<level>lvl.2</level>` (non-matching xp value and level)
+.. Test case: `<xp>559</xp>` `<xpValueByDay>90</xpValueByDay>` `<xpValueByWeek>60</xpValueByWeek>` +
+(Inconsistent xp values. All-time xp should never be smaller than this week's xp, which should never be smaller
+than today's xp.)
+.. Test case: `<numTaskCompleted>10</numTaskCompleted>` `<numTaskCompletedByDay>12</numTaskCompletedByDay>`
+`<numTaskCompletedByWeek>12</numTaskCompletedByWeek>` +
+(Inconsistent number of tasks completed. All-time number should never be smaller than this week's number, which
+should never be smaller than today's number.)
+.. Test case: `<nextDayBreakPoint>12-11-18 0000</nextDayBreakPoint>`
+`<nextWeekBreakPoint>19-11-18 0000</nextWeekBreakPoint>` +
+(Inconsistent date breakpoints. `nextWeekBreakPoint` should never be earlier than `nexDayBreakPoint`, it should never be
+more than 6 days later than nexDayBreakPoint.)
+.. Test case: `<xp>1000000001</xp>` +
+(The maximum integer value maintained by achievement record is 1000000000.)
+
+Expected: Due to the invalid data fields, the data file is considered corrupted and task manager is initialized with an
+empty data file. Task manager will be empty.
```

Comment:
Just to check, does this mean that both the tasks and acr will be reinitialised ?

Reply:
Yep! Task manager consists of task list, ach rec and game manager.

Reply:
Ok! sorry this tripped me up because there are two separate indicateTaskManagerChanged () and indicateACRUpdated(). Not sure if we want to view ACR and TM as separate. It's a composition relationship after I checked the code.